src/compiler/checker.ts

```diff
@@ -8980,7 +8980,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (result) {
                         if (result.pos !== -1 || result.end !== -1) {
                             if (result === nodes) {
-                                result = factory.createNodeArray(nodes, nodes.hasTrailingComma);
+                                result = factory.createNodeArray([...nodes], nodes.hasTrailingComma);

```

Comment:
I *believe* `slice()` tends to be a little faster than spreading since engines have to do shenanigans to support non-array iterables.

```suggestion
                                result = factory.createNodeArray(nodes.slice(), nodes.hasTrailingComma);
```